### PR TITLE
Add testing routine to run rtgsfit with text data files

### PR DESCRIPTION
**Pull request by FilipJankyTE**
- Changes into src/Makefile to save .so into lib

### CODE REVIEW

**Review by jbland64 — COMMENTED** [review]
* change absolute paths to relative i.e. try and make files user independent
* either include or remove commented out code

File: src/Makefile [hunk 1]
```diff
@@ -16,39 +16,37 @@ else
 	LDFLAGS = -shared -Wl,-Ofast -pthread
 endif
 
-OBJS = constants.o solve_tria.o gradient.o poisson_solver.o find_x_point.o rtgsfit.o 
-SHARED = $(patsubst %.o,%.so,$(OBJS))
-
+OBJS = constants.o solve_tria.o gradient.o poisson_solver.o find_x_point.o rtgsfit.o
+LIB = /home/filip.janky/rt-gsfit/rtgsfit/lib
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply src/Makefile:20]: I have done the test. It works. Great catch. Thank you

File: tests/makefile_test [hunk 2]
```diff
@@ -0,0 +1,35 @@
+CC = gcc
+CFLAGS += -fPIC -Wall -Warray-bounds -pg -std=gnu11
+MDSTOOL_SRC := ~/pcs/model/st40pcs/tools/src
+RTGSFIT := /home/filip.janky/rt-gsfit/rtgsfit/src
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply tests/makefile_test:4]: Relative path to pcs doesn't exist. As everyone has to set his/her own path. But currently, it is all at one place

File: tests/makefile_test [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+TE_INCLUDES := /home/filip.janky/pcs/model/ST40PCS/include/
+TE_OBJS := /home/filip.janky/pcs/model/ST40PCS/bin/
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply tests/makefile_test:6]: relative to selected pcs path

File: tests/makefile_test [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+OBJS = constants.o solve_tria.o gradient.o poisson_solver.o find_x_point.o rtgsfit.o
+SHARED = $(patsubst %.o,%.so,$(OBJS))
+
+MDSPLUS_ROOT := /usr/local/mdsplus
+MDSPLUS_LDFLAG := /usr/local/mdsplus/lib
+MDSPLUS_SYSLIB := -lMdsLib -lMdsShr -l
+SYSLIB := -lm
+
+MYLIB = -L/home/filip.janky/rt-gsfit/rtgsfit/lib
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply tests/makefile_test:15]: done. Nice catch

File: tests/rtgsfit_fileIO.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  char *fn_error = "../data/error.txt";
+  char character;
+
+  if (count_lines(fn_sensors_idx, &sensors_num) != 0) {
+    errorExit("Cannot count lines");
+  }
+  // if (N_MEAS != sensors_num) {
+  //   snprintf(error_msg, sizeof(error_msg), "Incorrect amount of sensors index %d %d\n", N_MEAS, sensors_num);
+  //   errorExit(error_msg);
+  // }
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply tests/rtgsfit_fileIO.c:27]: Good job. Thank you.

File: tests/rtgsfit_fileIO.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  if (get_int_from_file(fn_sensors_idx, N_MEAS, sensors_idx) != 0) {
+    errorExit("Cannot get array for coil sensors index.\n");
+  }
+  if (count_lines(fn_pf_coil_idx, &pf_coils_num) != 0) {
+    errorExit("Cannot count lines");
+  }
+  // if (N_COIL != pf_coils_num) {
+  //   snprintf(error_msg, sizeof(error_msg), "Incorrect amount of pf_coils %d %d\n", N_COIL, pf_coils_num);
+  //   errorExit(error_msg);
+  // }
```
[the remark this answers is not in the document]
Comment (FilipJankyTE) [reply tests/rtgsfit_fileIO.c:38]: Well spotted

**Pull request merged**